src/cpp/session/modules/rmarkdown/NotebookQueue.cpp

```diff
@@ -0,0 +1,560 @@
+/*
+ * NotebookQueue.cpp
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#include "SessionRmdNotebook.hpp"
+#include "NotebookQueue.hpp"
+#include "NotebookQueueUnit.hpp"
+#include "NotebookExec.hpp"
+#include "NotebookDocQueue.hpp"
+#include "NotebookCache.hpp"
+#include "NotebookAlternateEngines.hpp"
+
+#include "../../SessionClientEventService.hpp"
```

Comment:
Typically if we ever need to do `..` or `../..` for a header we just move it to the `include` directory.